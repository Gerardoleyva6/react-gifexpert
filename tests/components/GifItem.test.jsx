import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas en el nombre del coponente ', () => {

    const title = 'Gerardo'
    const url = 'https://gera.com/loquesea.jpg'

    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <GifItem title={title} url={url} /> )
        expect(container).toMatchSnapshot();
    })
    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={title} url={url} /> )
        //screen.debug()
        //expect(screen.getByRole('img').src).toBe(url)
        //expect(screen.getByRole('img').alt).toBe(title)
        const {src,alt}= screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })
    test('debe de mostrar el titulo en el coponente', () => {
        render( <GifItem title={title} url={url} /> )
        expect(screen.getByText(title)).toBeTruthy();
    })
})