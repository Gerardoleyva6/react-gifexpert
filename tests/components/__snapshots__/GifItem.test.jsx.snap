// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el nombre del coponente  debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gerardo"
      src="https://gera.com/loquesea.jpg"
    />
    <p>
      Gerardo
    </p>
  </div>
</div>
`;
